mesonbuild/backend/ninjabackend.py

```diff
@@ -3422,6 +3400,25 @@ def generate_prelink(self, target, obj_list):
         self.add_build(elem)
         return [prelink_name]
 
+    def get_build_rpath_args(self, target: build.BuildTarget, linker: T.Union[Compiler, StaticLinker]) -> T.List[str]:
+        if has_path_sep(target.name):
+            # Target names really should not have slashes in them, but
+            # unfortunately we did not check for that and some downstream projects
+            # now have them. Once slashes are forbidden, remove this bit.
+            target_slashname_workaround_dir = os.path.join(
+                os.path.dirname(target.name),
+                self.get_target_dir(target))
```

Comment:
This should probably be:

```suggestion
            target_slashname_workaround_dir = os.path.join(os.path.dirname(target.name),
                                                           self.get_target_dir(target))
```

(just like the code it was moved from right above)